### PR TITLE
DOC: Update ndimage.morphology.distance_transform_*

### PR DESCRIPTION
**Pull request by terrdavis**
#### Reference issue
#12118

#### What does this implement/fix?
For the following functions in scipy.ndimage.morphology: 
 - distance_transform_edt
 - distance_transform_cdt
 - distance_transform_bf

Fix inconsistencies and inaccuracies in the documentation.

### CODE REVIEW

File: scipy/ndimage/morphology.py [hunk 1]
```diff
@@ -2075,33 +2108,46 @@ def distance_transform_edt(input, sampling=None, return_distances=True,
...
-    return_distances : bool, optional
-        Whether to return distance matrix. At least one of
-        return_distances/return_indices must be True. Default is True.
-    return_indices : bool, optional
-        Whether to return indices matrix. Default is False.
-    distances : ndarray, optional
-        Used for output of distance array, must be of type float64.
-    indices : ndarray, optional
-        Used for output of indices, must be of type int32.
+    return_distances, return_indices : bool, optional
```
Comment (sethtroisi) [inline scipy/ndimage/morphology.py:2122]: Not sure how I feel about combining these on one line.

File: scipy/ndimage/morphology.py [hunk 2]
```diff
@@ -2075,33 +2108,46 @@ def distance_transform_edt(input, sampling=None, return_distances=True,
...
+        
+        Its shape must be `(input.ndim,) + input.shape`.
 
     Returns
     -------
-    distance_transform_edt : ndarray or list of ndarrays
-        Either distance matrix, index matrix, or a list of the two,
-        depending on `return_x` flags and `distance` and `indices`
-        input parameters.
+    distances : float64 ndarray, optional
```
Comment (sethtroisi) [inline scipy/ndimage/morphology.py:2142]: It would be nice to retain the information that returns always has size 1

**Review by sethtroisi — COMMENTED** [review]
My eyes jumped about but overall this looks reasonable

File: scipy/ndimage/morphology.py [hunk 3]
```diff
@@ -1827,45 +1827,53 @@ def distance_transform_bf(input, metric="euclidean", sampling=None,
...
-
-        The default is False.
-    distances : float64 ndarray, optional
-        Optional output array to hold distances (if `return_distances` is
-        True).
-    indices : int64 ndarray, optional
-        Optional output array to hold indices (if `return_indices` is True).
+    metric : {'euclidean', 'taxicab', 'chessboard'}, optional
+        'cityblock' and 'manhattan' are also valid, and map to 'taxicab'.
+
```
Comment (rgommers) [inline scipy/ndimage/morphology.py:1838]: while you're at it, can you please remove all the blank lines like this one in parameter descriptions?

File: scipy/ndimage/morphology.py (hunk 1 above)
Comment (rgommers) [reply scipy/ndimage/morphology.py:2122]: agreed, this is better as separate descriptions

**Comment by rgommers** [comment]
thanks for working on this @terrdavis. `THANKS.txt` already has a merge conflict, could you resolve that by rebasing or merging master into your branch?

File: scipy/ndimage/morphology.py (hunk 2 above)
Comment (terrdavis) [reply scipy/ndimage/morphology.py:2142]: As in, always the same size as the input?

**Comment by terrdavis** [comment]
Changed to a draft, I'm going to propose an API change on the mailing list. That change would supersede this PR if green-lit.

File: scipy/ndimage/morphology.py [hunk 4]
```diff
@@ -1827,45 +1827,49 @@ def distance_transform_bf(input, metric="euclidean", sampling=None,
...
-        Indices array if `return_indices` is True.
+    distances : ndarray, optional
+        The calculated distance transform. Returned only when
+        `return_distances` is True and `distances` is not supplied.
+        It will have the same shape as the input array.
+    indices : int32 ndarray, optional
+        The calculated feature transform. It has an input-shaped array for each 
+        dimension of the input. See distance_transform_edt documentation for an 
+        example.
+        Returned only when `return_indices` is True and `indicies` is not
```
Comment (adeak) [inline scipy/ndimage/morphology.py:1871]: Minor nitpick: it should spell `indices` everywhere, not `indicies` :)

Comment (terrdavis) [reply scipy/ndimage/morphology.py]: Thank you!

**Comment by terrdavis** [comment]
I'm marking this as ready for review again. I'll open another PR for the other change since I'll need to add tests, which will take more time.

**Review by rgommers — APPROVED** [review]
This looks good, let's get it in. Thanks @terrdavis!

**Pull request merged**